From be8e1c08c453051d4e8eaaa62698356f9679d5a1 Mon Sep 17 00:00:00 2001
From: npalanivel <npalanivel@maxlinear.com>
Date: Fri, 26 Jul 2024 14:11:56 +0530
Subject: [PATCH] WLANRTSYS-82145 : [Workaround] Made changes to hardcode HUNT
 AND PECK method to avoid client disconnection.

To-Do: remove these changes once veriwave supports hashing.
---
 hostapd/mxl_config.c | 10 ++++++++++
 src/ap/mxl_config.h  |  1 +
 2 files changed, 11 insertions(+)

diff --git a/hostapd/mxl_config.c b/hostapd/mxl_config.c
index eb542700a..58e79e4b6 100644
--- a/hostapd/mxl_config.c
+++ b/hostapd/mxl_config.c
@@ -1916,6 +1916,8 @@ int mxl_hostapd_config_fill (struct hostapd_config *conf,
 			return 0;
 		}
 		mxl_conf->zwdfs_debug_chan = val;
+	} else if (os_strcmp(buf, "veriwave_test") == 0) {
+		mxl_bss_conf->veriwave_test = atoi(pos);
 	} else
 		return 0;
 
@@ -2037,6 +2039,14 @@ void mxl_hostapd_config_free(struct hostapd_config *conf)
 }
 
 int mxl_hostapd_config_check_bss_6g(struct hostapd_bss_config *bss) {
+	/*
+	 * WLANRTSYS-82145 - workaround for veriwave test suite
+	 * To-Do: remove these changes once veriwave supports hashing.
+	 */
+	if (bss->mxl_bss_conf.veriwave_test) {
+		bss->sae_pwe = SAE_PWE_HUNT_AND_PECK;
+	}
+
 	if (bss->wpa_key_mgmt & (WPA_KEY_MGMT_IEEE8021X | WPA_KEY_MGMT_IEEE8021X_SHA256)) {
 		wpa_printf(MSG_ERROR, "Invalid AKM suite for 6 GHz");
 		return -1;
diff --git a/src/ap/mxl_config.h b/src/ap/mxl_config.h
index 79e27c919..2d1105000 100644
--- a/src/ap/mxl_config.h
+++ b/src/ap/mxl_config.h
@@ -404,6 +404,7 @@ struct mxl_hostapd_bss_config {
 	bool mlo_t2lm_support; /* tid to link mapping negotiation support */
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 	int rnr_auto_update;
+	bool veriwave_test;
 };
 
 struct hostapd_config;
-- 
2.43.0

